lib/summarize-all.js

```diff
@@ -3,20 +3,23 @@
 const path = require('path');
 const summarize = require('./summarize');
 const walk = require('walk-sync');
+const workerpool = require('workerpool');
+
+const pool = workerpool.pool(path.join(__dirname, 'summarize-worker.js'));
 
 /**
  * Walk over all json files generated from broccoli-concat and write *.out.json containing size summary (uglified/compressed)
  * of all referenced files
  *
  * @param concatStatsForPath
  * @param cb
+ * @return {Promise}
  */
-module.exports = function summarizeAll(concatStatsForPath, cb) {
+module.exports = function summarizeAll(concatStatsForPath) {
   const entries = walk.entries(concatStatsForPath, { globs: ['*.json'], ignore: ['*.out.json'] });
-  entries.forEach(entry => {
-    if (cb) {
-      cb(entry);
-    }
-    summarize(path.join(concatStatsForPath, entry.relativePath));
-  });
+  return Promise.all(
+    entries.map(entry => pool.exec('summarize', [path.join(concatStatsForPath, entry.relativePath)]))
+  )
+    .catch((err) => console.error(err))
```

Comment:
If there are errors, how do we ensure the process exists with a non-zero status code?

Reply:
@stefanpenner I removed the catch, so this function will properly return a rejected promise in case of an error. The error logging happens in cli.js now, where it belongs, and it returns an exit code of 1. So that should be addressed now!